### DISCUSSION TITLE
question about obj files containing multiple objects

### DISCUSSION THREAD

**Discussion by jg10545** [facebookresearch/pytorch3d#1694] [General]
Hey, I've got a possibly-naive question I'd love some insight on from any of the experts here. This may be related to #1572 but I'm not 100% sure.

Following the ["Render a textured mesh"](https://pytorch3d.org/tutorials/render_textured_meshes) tutorial I can reproduce the cow picture pretty easily:

![just_the_cow](https://github.com/facebookresearch/pytorch3d/assets/42722768/af2930b1-bc35-4f12-a86f-4d68eaeb83d9)

If I load the cow `.obj` file into Blender, add another object, and export it to a new `.obj` file:

![cow_and_slab_in_blender](https://github.com/facebookresearch/pytorch3d/assets/42722768/710be45c-64d0-426d-9b16-380fc0199b65)

When I run that same tutorial code with the new mesh file, PyTorch3D wraps the cylinder with the cow texture:

![cow_slab_pytorch3d](https://github.com/facebookresearch/pytorch3d/assets/42722768/136501f7-8366-4aca-80d8-44e099b3157a)

This doesn't appear to be an error in the wavefront files; the `.mtl` shows a material but not a texture for the cylinder (and if I import it back into Blender it's still green).

Is this expected behavior? Am I missing something obvious? Any workarounds?

Thanks in advance for any insight!

**Comment by bottler** [comment]
We only support a single texture uv-map per mesh. Loading an obj file produces one mesh in PyTorch3D. I don't know exactly what has happened here: e.g. does the OBJ file know that it contains two mesh pieces, only one of which is textured, and PyTorch3D ignores that, or does blender save something somehow with a single texture all over. 

If you want the cylinder to be green in PyTorch3D, perhaps this would work:
- export just the cylinder to OBJ. 
- Load the cylinder and cow obj files into pytorch3d, producing two Meshes.
- Assign a new TexturesUV object with a constant green color to the cylinder mesh.
- Use join_meshes_as_scene to amalgamate the two meshes, which will automatically amalgamate their textures into a single larger texture map.

**Comment by jg10545** [comment]
@bottler thank you, the `join_meshes_as_scene()` function worked great!!
